### PR TITLE
Bug 1263562 - users without projects should get default ctx when logging in

### PR DESCRIPTION
**Pull request by fabianofranz**

### CODE REVIEW

**Comment by fabianofranz** [comment]
[test]

**Comment by liggitt** [comment]
what happens today? the default namespace will certainly not work for them... not sure what we get by setting them to it

**Comment by fabianofranz** [comment]
@liggitt today it's worse - you get the last namespace used by the last user logged in before the current. So *ffranz* is logged in using namespace *test*, then *liggitt* logs is, has no namespace but his context point to *test*.

**Comment by fabianofranz** [comment]
s/has no namespace/has no projects

**Comment by fabianofranz** [comment]
If the user has no projects we still have to save something in the namespace field of the current context, and the last one used is definitely not good. Maybe just empty string instead of *default*?

**Comment by liggitt** [comment]
I thought it was the empty string

**Comment by liggitt** [comment]
I would prefer empty, though I'm not sure what that breaks

**Comment by fabianofranz** [comment]
@liggitt most likely doesn't break anything, but I think empty will end up trying to use *default*. But let me check.

**Comment by fabianofranz** [comment]
@liggitt It ends up using *default* if you try to do anything like `oc get`, but empty string seems to reflect it better in the config file (you don't get the `namespace: ` line). Updated.

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -222,6 +222,9 @@ oc login --server=${KUBERNETES_MASTER}/ --certificate-authority="${MASTER_CONFIG
 # create a new project
 oc new-project project-foo --display-name="my project" --description="boring project description"
 [ "$(oc project | grep 'Using project "project-foo"')" ]
+# new user should get default context
+oc login --server=${KUBERNETES_MASTER}/ --certificate-authority="${MASTER_CONFIG_DIR}/ca.crt" -u new-and-unknown-user -p anything
```
Comment (liggitt) [inline hack/test-cmd.sh]: why the trailing slash on `${KUBERNETES_MASTER}/`?

Comment (fabianofranz) [inline hack/test-cmd.sh]: Ha, that came when I copied a previous line.  :)

Comment (fabianofranz) [inline hack/test-cmd.sh]: Removed from this and other occurrences.

**Comment by fabianofranz** [comment]
@liggitt should be good to go.

**Comment by liggitt** [comment]
lgtm, @deads2k can you double check?

**Comment by deads2k** [comment]
https://github.com/openshift/origin/issues/4948

**Comment by deads2k** [comment]
Part of this problem was that when I logged in as "bob", created project "bobs-project" and then logged back in "system:admin", I ended up as "system:admin" in "bobs-project" right?  Can you make sure that doesn't happen in `test-cmd.sh`?

**Comment by fabianofranz** [comment]
@deads2k something like that, it only happens when you switch to an user that doesn't have projects. E.g. you logged is as "bob", created "bobs-project" then you log in as "user-without-projects" and stay in "bobs-project". I added a test for that in `test-cmd.sh`.

**Comment by deads2k** [comment]
lgtm [merge]

**Pull request merged**